src/mscorlib/src/System/Diagnostics/Eventing/EventPipeEventProvider.cs

```diff
@@ -49,11 +51,40 @@ uint IEventProvider.EventUnregister(long registrationHandle)
         unsafe int IEventProvider.EventWriteTransferWrapper(
             long registrationHandle,
             ref EventDescriptor eventDescriptor,
+            IntPtr eventHandle,
             Guid* activityId,
             Guid* relatedActivityId,
             int userDataCount,
             EventProvider.EventData* userData)
         {
+            uint eventID = (uint)eventDescriptor.EventId;
+            if(eventID != 0 && eventHandle != IntPtr.Zero)
+            {
+                if (userDataCount == 0)
+                {
+                    EventPipeInternal.WriteEvent(eventHandle, eventID, null, 0);
+                    return 0;
+                }
+
+                uint length = 0;
+                for (int i = 0; i < userDataCount; i++)
+                {
+                    length += userData[i].Size;
+                }
+
+                byte[] data = new byte[length];
```

Comment:
We don't want any allocations on the normal logging path.  This is also another buffer copy that we can avoid.     
We can leave this in the VERY short term (today), but I would like to change thing so that

EventPipeInternal.WriteEvent takes a int userDataCount, and  EventProvider.EventData* userData.   instead of a byte* and length.    You should simply pass along these until the real work happens in EventPipeBufferManager.WriteEvent, and then do the one and only copy needed there.